WordPress/Sniffs/Theme/NoFaviconSniff.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Check for hardcoded favicons instead of using core implementation.
+ *
+ * @link    https://make.wordpress.org/themes/handbook/review/required/#core-functionality-and-features
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NoFaviconSniff implements PHP_CodeSniffer_Sniff {
+
+	const REGEX_TEMPLATE = '` (?:%s)`i';
+
+	const REGEX_ATTR_TEMPLATE = '%1$s=[\'"](?:%2$s)[\'"]';
+
+	/**
+	 * List of link and meta attributes that are blacklisted.
+	 *
+	 * @var array
+	 */
+	protected $attribute_blacklist = array(
+		'rel' => array(
+			'icon',
+			'shortcut icon',
+			'bookmark icon',
+			'apple-touch-icon',
+			'apple-touch-icon-precomposed',
+		),
+		'name' => array(
+			'msapplication-config',
+			'msapplication-TileImage',
+			'msapplication-square70x70logo',
+			'msapplication-square150x150logo',
+			'msapplication-wide310x150logo',
+			'msapplication-square310x310logo',
+		),
+	);
+
+	/**
+	 * The regex to catch the blacklisted attributes.
+	 *
+	 * @var string
+	 */
+	protected $favicon_regex;
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		$regex_parts = array();
```

Comment:
May be add a comment above this line `// Build the regex to be used only once.` ?